tests/unit/plugins/v2/test_cmake.py

```diff
@@ -64,8 +64,8 @@ class Options:
             Equals(
                 [
                     'cmake "${SNAPCRAFT_PART_SRC_WORK}"',
-                    'cmake --build . -- -j"${SNAPCRAFT_PARALLEL_BUILD_COUNT}"',
-                    'cmake --build . --target install -- DESTDIR="${SNAPCRAFT_PART_INSTALL}"',
+                    'cmake --build . -j"${SNAPCRAFT_PARALLEL_BUILD_COUNT}"',
+                    'cmake --install . --prefix="${SNAPCRAFT_PART_INSTALL}"',
```

Comment:
I could be wrong, but I don't think this is the correct use of `--prefix`.  While it may certainly install there as a result, it seems likely to have other potential side effects...

Reply:
It seems, that you are right. `cmake --install X --prefix Y` is equivalent to `cmake -DCMAKE_INSTALL_PREFIX=Y -P X/cmake_install.cmake`, and setting CMAKE_INSTALL_PREFIX should not be used for packaging purposes.

DESTDIR is the way to go here, but when I try to use Ninja generator, I get an error:
```
cmake --build . --target install -- DESTDIR=XYZ                        
[0/2] Re-checking globbed directories...
ninja: error: unknown target 'DESTDIR=XYZ'
```
Ninja supports DESTDIR using environment variable, while plain-old make does not, and only supports it using a CLI argument.

I think, that the best way to fix Ninja builds is to add a new parameter to plugin: `cmake-generator`, that (if set) passes `-G` parameter to first `cmake` invocation and then:
- if `cmake-generator` contains "Ninja" (case-sensitive), then DESTDIR is passed using env and `cmake --install` is invoked (it is much faster, than cmake --build install).
- else pass DESTDIR as `cmake --build` argument.

This way this plugin will gain support of Ninja build system, which is much faster.